src/net_processing.cpp

```diff
@@ -5076,6 +5072,24 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
 
     MaybeSendAddr(*pto, *peer, current_time);
 
+    // Delay sending SENDHEADERS (BIP 130) until we're done with an
+    // initial-headers-sync with this peer. Receiving headers announcements for
+    // new blocks while trying to sync their headers chain is problematic,
+    // because of the state tracking done.
+    if (!peer->m_sent_sendheaders && pto->GetCommonVersion() >= SENDHEADERS_VERSION) {
```

Comment:
2444ce2f8cb1330ba1dc88b7e5407a3efcfd9aeb The above `MaybeSendPing` and `MaybeSendAddr`  suggests we should put this in a helper function `MaybeSendSendHeaders()`

Reply:
Done. :)